### PR TITLE
options/posix: implement pthread_detach()

### PR DESCRIPTION
**Pull request by Geertiebear**

### CODE REVIEW

File: options/posix/generic/pthread-stubs.cpp [hunk 1]
```diff
@@ -214,9 +219,19 @@ int pthread_join(pthread_t thread, void **ret) {
...
 
-int pthread_detach(pthread_t) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int pthread_detach(pthread_t thread) {
+	auto tcb = reinterpret_cast<Tcb*>(thread);
+	if (!(tcb->selfPointer == tcb))
+		return ESRCH;
+	if (!tcb->isJoinable)
+		return EINVAL;
```
Comment (avdgrinten) [inline options/posix/generic/pthread-stubs.cpp:227]: This needs to do an `__atomic_load_n` (same for all other accesses to `isJoinable`).

File: options/posix/generic/pthread-stubs.cpp [hunk 2]
```diff
@@ -214,9 +219,19 @@ int pthread_join(pthread_t thread, void **ret) {
 	return 0;
 }
 
-int pthread_detach(pthread_t) {
-	__ensure(!"Not implemented");
-	__builtin_unreachable();
+int pthread_detach(pthread_t thread) {
+	auto tcb = reinterpret_cast<Tcb*>(thread);
+	if (!(tcb->selfPointer == tcb))
+		return ESRCH;
```
Comment (avdgrinten) [inline options/posix/generic/pthread-stubs.cpp:225]: Is this *correct*? Why is it done?

Comment (Geertiebear) [reply options/posix/generic/pthread-stubs.cpp:225]: A sanity check.

**Review by avdgrinten — APPROVED** [review]
LGTM.

**Pull request merged**